server/mediaserver.go

```diff
@@ -759,10 +754,30 @@ func (s *LivepeerServer) HandlePush(w http.ResponseWriter, r *http.Request) {
 			http.Error(w, httpErr, http.StatusInternalServerError)
 			return
 		}
-		st := stream.NewBasicRTMPVideoStream(appData)
-		params := streamParams(st)
+		params := streamParams(appData)
 		params.Resolution = r.Header.Get("Content-Resolution")
 		params.Format = format
+		if params.ExternalID != "" {
+			// Pre-existing connection found for this new stream with the same underlying manifestID
+			var oldStreamID core.ManifestID
+			s.connectionLock.RLock()
+			for k, v := range s.internalManifests {
+				if v == params.ManifestID {
+					oldStreamID = k
+					break
+				}
+			}
+			s.connectionLock.RUnlock()
+			if oldStreamID != "" && params.ExternalID != oldStreamID {
+				// Close the old connection, and open a new one
+				// TODO try to re-use old HLS playlist?
+				glog.Warningf("Ending streamID=%v as new streamID=%s with same manifestID=%s has arrived",
+					oldStreamID, params.ExternalID, params.ManifestID)
+				removeRTMPStream(s, oldStreamID)
+				mid = params.ExternalID
```

Comment:
At that point `mid` is equal to the `params.ExternalID` anyway. Probably we shouldn't have added `ExternalID` field to the `params` - it's the same value parsed from the URL. We could just check here if 
```go
mid != params.ManifestID && s.rtmpConnections[params.ManifestID] != nil && s.internalManifests[mid] == nil {
```
instead of 
```go
if params.ExternalID != "" {
```